src/Client.php

```diff
@@ -298,10 +298,21 @@ protected function paramsToQueryString(array $params = [])
             if (is_array($value)) {
                 // recursion case
 
-                $result_string = $this->paramsToQueryString($value);
-                if (!empty($result_string)) {
-                    $query_string[] = $result_string;
+                if ( empty( $value ) ) {
```

Comment:
Reading the GeoNames doc, looks like names without values (i.e. `url?name`) are not supported. Am I missing something?

Reply:
No, I have not checked the doc in this respect. What shall we do with an empty value then? Drop it or throw an exception?

My suggestion would be to silently drop it as it is an empty array, not an empty string or zero.

Reply:
That sounds like a good idea.